app/services/password_crypt.rb

```diff
@@ -1,9 +1,11 @@
 class PasswordCrypt
-  ALGORITHMS = {'MD5' => '$1$', 'SHA256' => '$5$', 'SHA512' => '$6$'}
+  ALGORITHMS = {'MD5' => '$1$', 'SHA256' => '$5$', 'SHA512' => '$6$', 'Base64' => ''}
 
   def self.passw_crypt(passwd, hash_alg = 'MD5')
     raise Foreman::Exception.new(N_("Unsupported password hash function '%s'"), hash_alg) unless ALGORITHMS.has_key?(hash_alg)
-    passwd.crypt("#{ALGORITHMS[hash_alg]}#{SecureRandom.base64(6)}")
+    require 'base64'
+    # strict_base64 isn't available until Ruby 1.9.2
```

Comment:
We don't support < 1.9.3 currently, so feel free to remove any alternative options :smile: